discovery/syncer.go

```diff
@@ -1274,11 +1303,23 @@ func (g *GossipSyncer) FilterGossipMsgs(msgs ...msgWithSenders) {
 
 // ProcessQueryMsg is used by outside callers to pass new channel time series
 // queries to the internal processing goroutine.
-func (g *GossipSyncer) ProcessQueryMsg(msg lnwire.Message, peerQuit <-chan struct{}) {
+func (g *GossipSyncer) ProcessQueryMsg(msg lnwire.Message, peerQuit <-chan struct{}) error {
 	var msgChan chan lnwire.Message
 	switch msg.(type) {
 	case *lnwire.QueryChannelRange, *lnwire.QueryShortChanIDs:
 		msgChan = g.queryMsgs
+
+	// Reply messages should only be expected in states where we're waiting
+	// for a reply.
+	case *lnwire.ReplyChannelRange, *lnwire.ReplyShortChanIDsEnd:
```

Comment:
Hm, this change seems independent to the rest of the commit? Also not clear to me why it is needed.

Reply:
Just something I noticed while working on the PR, things can block here for a bit if the state machine's not expecting a reply.

Reply:
ok, would be nice in its own commit with an explanation